packages/pages/src/dev/server/ssr/getPluginFileMap.ts

```diff
@@ -0,0 +1,44 @@
+import path from "path";
+import fs from "fs";
+
+const PLUGIN_FILE_MAP_PATH = path.resolve(
+  ".artifact-output/pluginFileMap.json"
+);
+
+/** The type of event for the serverless function. */
+export enum EventType {
+  /** The function is in an error state. */
+  UNKNOWN_EVENT = 0,
+  /** A function that is invoked on a URL change event. */
+  ON_URL_CHANGE = 1,
+  /** A function that is invoked on a page generate event. */
+  ON_PAGE_GENERATE = 2,
+  /** An API (or HTTP) function. */
+  API = 3,
+}
+
+/** Source data associated with a serverless function. */
+export type PluginSource = {
```

Comment:
Could we call this `FunctionSource` or `ServerlessFunctionSource`? I think it's bad enough that we use the terms Plugin and Serverless Function interchangeably in the platform. In our code, I'd like us to pick one term.